dotnet/runtime
# Intrinsify typeof(T).IsPrimitive

Closes https://github.com/dotnet/runtime/issues/95929

```cs
bool Test<T>() => typeof(T).IsPrimitive;
```
Codegen diff for e.g. `Test<int>` and `Test<string>`:
```diff
; Assembly listing for method P:Test[int]():ubyte (FullOpts)
-      sub      rsp, 40
-      mov      rcx, 0x27E800054B0      ; 'System.Int32'
-      call     System.RuntimeTypeHandle:GetCorElementType(System.RuntimeType):ubyte
-      mov      ecx, 1
-      shlx     eax, ecx, eax
-      test     eax, 0x3003FFC
-      setne    al
-      movzx    rax, al
-      add      rsp, 40
+      mov      eax, 1
       ret      

; Assembly listing for method P:Test[System.__Canon]():ubyte (FullOpts)
-      sub      rsp, 40
-      mov      qword ptr [rsp+0x20], rcx
-      mov      rcx, qword ptr [rcx+0x38]
-      mov      rcx, qword ptr [rcx]
-      call     CORINFO_HELP_TYPEHANDLE_TO_RUNTIMETYPE
-      mov      rcx, rax
-      call     System.RuntimeTypeHandle:GetCorElementType(System.RuntimeType):ubyte
-      mov      ecx, 1
-      shlx     eax, ecx, eax
-      test     eax, 0x3003FFC
-      setne    al
-      movzx    rax, al
-      add      rsp, 40
+      xor      eax, eax
       ret
```

Should allow us to avoid listing all primitives by hands for perf reasons (e.g. [here](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/Constants.cs#L46-L58))

will add tests.
